autorest/azure/azure.go

```diff
@@ -44,20 +44,37 @@ const (
 
 // ServiceError encapsulates the error response from an Azure service.
 type ServiceError struct {
-	Code    string         `json:"code"`
-	Message string         `json:"message"`
-	Details *[]interface{} `json:"details"`
+	Code       string         `json:"code"`
+	Message    string         `json:"message"`
+	Target     *string        `json:"target"`
+	Details    *[]interface{} `json:"details"`
+	InnerError *interface{}   `json:"innererror"`
```

Comment:
For JSON objects the type should be map[string]interface{} (and it doesn't need to be a pointer to map).